File: toki/belajar/dasar/P59Wartel.py
buku = {}
n,q = map(int, input().strip().split())
for i in range(n):
    nama,no = input().strip().split()
    buku[nama] = no
for i in range(q):
    nama = input()
    if nama in buku.keys():
        print(buku[nama])
    else:
        print("NIHIL")

File: toki/belajar/dasar/P59Wartel.cpp
#include <iostream>
#include <map>
using namespace std;

typedef map<string, string> Dict;
typedef Dict::const_iterator It;

int main()
{
   Dict buku;
   int n,q;
   string nama,no,tampil;
   cin>>n>>q;
   for (int i=0; i<n; i++)
   {
   		cin>>nama>>no;
   		buku[nama] = no;
   }
   for (int i=0; i<q; i++)
   {
   		tampil="NIHIL";
   		cin>>nama;
   		for (It it(buku.begin()); it != buku.end(); ++it)
   		{
      		string i(it->first); 
      		if (nama == it->first) 
			{
				tampil = buku[nama];
				break;
			}
   		}
   		cout<<tampil<<"\n";
   }
   return 0;
}
